language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1217,4 +1222,38 @@ public static Boolean isInFunctionCallParameterContext(PositionedOperationContex
                 && (!node.closeParenToken().isMissing())
                 && (cursorPosition <= node.closeParenToken().textRange().startOffset());
     }
+
+    /**
+     * Check if the symbol is a class symbol with self as the name.
+     *
+     * @param symbol  Symbol
+     * @param context PositionedOperationContext
+     * @param enclosedModuleMember ModuleMemberDeclarationNode
+     * @return {@link Boolean} whether the symbol is a self class symbol.
+     */
+    public static boolean isSelfClassSymbol(Symbol symbol, PositionedOperationContext context,
+                                            ModuleMemberDeclarationNode enclosedModuleMember) {
+
+        Optional<String> name = symbol.getName();
+
+        if (enclosedModuleMember != null) {
+            if (enclosedModuleMember.kind() != SyntaxKind.CLASS_DEFINITION || symbol.kind() != SymbolKind.VARIABLE
+                    || name.isEmpty() || !name.get().equals(SELF_KW)) {
+                return false;
+            }
+        }
+
+        Optional<Symbol> memberSymbol = context.workspace().semanticModel(context.filePath())
+                .get().symbol(enclosedModuleMember);
```

Comment:
Need to check if semantic model is present (in optional) before invoking `get()` on the optional. We can rewrite this as:
```suggestion
                .flatMap(semanticModel -> semanticModel.symbol(enclosedModuleMember));
```